### PR TITLE
Log out of Link after transaction completes

### PR DESCRIPTION
**Pull request by davidme-stripe**
## Summary
Invalidate the client secret by calling `log_out` immediately after the transaction completes.

Remove a bit of related unused code from Link v1.

## Testing
* CI end-to-end tests, manual testing of Link inline signup.
* Bring the ConsumerSessionsTests to test this scenario, ensure the token no longer works after logout.

### CODE REVIEW

File: Stripe/StripeiOSTests/ConsumerSessionTests.swift [hunk 1]
```diff
@@ -139,8 +81,7 @@ class ConsumerSessionTests: XCTestCase {
 
         ConsumerSession.lookupSession(
             for: "mobile-payments-sdk-ci+not-a-consumer@stripe.com",
-            with: apiClient,
-            cookieStore: cookieStore
```
Comment (wooj-stripe) [inline Stripe/StripeiOSTests/ConsumerSessionTests.swift:143]: nit: probably should be: `not-a-consumer+\(UUID())`

File: Stripe/StripeiOSTests/ConsumerSessionTests.swift [hunk 2]
```diff
@@ -175,9 +116,8 @@ class ConsumerSessionTests: XCTestCase {
             phoneNumber: "+13105551234",
             legalName: nil,
             countryCode: "US",
-            consentAction: nil,
-            with: apiClient,
-            cookieStore: cookieStore
+            consentAction: "clicked_checkbox_nospm_mobile_v0",
```
Comment (wooj-stripe) [inline Stripe/StripeiOSTests/ConsumerSessionTests.swift:119]: use: `PaymentSheetLinkAccount.ConsentAction.checkbox_v0.rawValue` ?
though, this makes me think the function should accept an enum vs a string

File: Stripe/StripeiOSTests/ConsumerSessionTests.swift [hunk 3]
```diff
@@ -223,18 +164,39 @@ class ConsumerSessionTests: XCTestCase {
...
+                                case .failure(let error):
+
+                                    guard let stripeError = error as? StripeError,
+                                        case let .apiError(stripeAPIError) = stripeError else {
+                                        XCTFail("Received unexpected error response")
+                                        return
+                                    }
+                                    XCTAssertEqual(stripeAPIError.code, "consumer_session_credentials_invalid")
+                                }
+                                useDetailsAfterLogoutExpectation.fulfill()
```
Comment (wooj-stripe) [inline Stripe/StripeiOSTests/ConsumerSessionTests.swift:194]: Seems like this fulfill should be right under `XCTAssertEqual(stripeAPIError.code`, or we can fulfill prior to asserting.

Comment (davidme-stripe) [reply Stripe/StripeiOSTests/ConsumerSessionTests.swift:194]: If we fulfill prior to asserting, I think we have a chance of the main thread ending the test as successful before the assertion is checked. I've had to fix that issue in our other tests before.

Comment (wooj-stripe) [reply Stripe/StripeiOSTests/ConsumerSessionTests.swift:194]: Sorry, i messed up here.. I was thinking something that was just wrong. my bad!

**Review by wooj-stripe — APPROVED** [review]

**Pull request merged**